### DISCUSSION TITLE
What is `minimalMode` in next-server?

### DISCUSSION THREAD

**Discussion by genetschneider** [vercel/next.js#29801] [Help] [answered]
Usage of [`minimalMode`](https://github.com/vercel/next.js/blob/canary/packages/next/server/next-server.ts#L218) can be seen throughout the codebase. One of the key features (side-effects) of this flag seems to be the fact that next-server will [throw exceptions](https://github.com/vercel/next.js/blob/canary/packages/next/server/next-server.ts#L492) instead of sending "Internal Server Error" to the client. One of the benefits of this is to catch errors in case of custom server and introduce proper logging instead of relying on `console.error`. 

What does this flag do and is it safe to use?

**Comment by ijjk** [comment, marked as answer]
Minimal mode is an experimental config being tested in fully featured Next.js environments like Vercel that have an edge network to handle specific parts of requests that `next-server` would normally handle. I would not recommend using this config at this point as it is undocumented and can change at any time.
